### DISCUSSION TITLE
Annotated[BaseModelClass, Form()] won't work

### DISCUSSION THREAD

**Discussion by NathanAP** [fastapi/fastapi#12225] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

I'm not sure this is about the `0.113.0` (and `0.114.0`) changes, but trying this:

```python
from fastapi import APIRouter, Form
from pydantic import BaseModel
from typing import Annotated

class RequestBody(BaseModel):
    path: str

@router.post("/blabla/")
async def local_searcher(body: RequestBody):
   ...
```

leads me to `422` error everytime (through postman, starlette request, etc), so I tried to intercept this error doing: 

```python

@app.exception_handler(RequestValidationError)
async def validation_exception_handler(request: Request, exc: RequestValidationError):
   print(await request.body())
   print(exc)
```

and it printed something like: 
```
...
b'{ "path": "/some/path" }'
[{'type': 'missing', 'loc': ('body', 'path'), 'msg': 'Field required', ... }]
...
```

### Description

This is basically a copy/paste of [this](https://github.com/fastapi/fastapi/issues/12198#issuecomment-2353618319). 

I always used something like:

```python
async def local_searcher(body: RequestBody):
  ...
``` 

but with `Annotated[RequestBody, Form()]` I cannot. What am I doing wrong?

### Operating System

Linux

### FastAPI Version

0.115.0

### Pydantic Version

2.9.2

### Python Version

3.12.6

**Comment by sinisaos** [comment]
@NathanAP Sorry, but in my case your example works correctly in `fastapi>=0.113.0`. Your example with `Annotated[RequestBody, Form()]`

```python
from fastapi import FastAPI, Form
from pydantic import BaseModel
from typing import Annotated


class RequestBody(BaseModel):
    path: str


app = FastAPI()


@app.post("/blabla/")
async def local_searcher(body: Annotated[RequestBody, Form()]):
    return body
```
Result is:

```bash
> curl --form path=abcd http://127.0.0.1:8000/blabla/
{"path":"abcd"}
```
`Swagger UI` also works. Sorry if I missed your point

**Reply by NathanAP** [reply]
Sorry, might be related to something else, really.

I tried testing this code:

```python

from fastapi import FastAPI, Form
from pydantic import BaseModel
from typing import Annotated

class Path(BaseModel):
    url: str

class RequestBody(BaseModel):
    path: Path

app = FastAPI()

@app.post("/blabla/")
async def local_searcher(body: Annotated[RequestBody, Form()]):
    return body
```

When I send via Postman:

```json
{
    "path": {
        "url": "www.example.com"
    }
}
```

I get 422.

**Reply by sinisaos** [reply, marked as answer]
@NathanAP It's a different story because you're using **nested** models that **don't work.** I see other people reporting this problem in `Query()` as well. You probably have to wait for the FastAPI team to sort it out. Good luck.
